pygls/server.py

```diff
@@ -426,6 +433,37 @@ def show_message_log(self, message, msg_type=MessageType.Log) -> None:
         """Sends message to the client's output channel."""
         self.lsp.show_message_log(message, msg_type)
 
+    def _report_server_error(self, error: Exception, source: Union[PyglsError, JsonRpcException]):
+        # Prevent recursive error reporting
+        try:
+            self.report_server_error(error, source)
+        except Exception:
+            logger.warning("Failed to report error to client")
+
+    def report_server_error(self, error: Exception, source: Union[PyglsError, JsonRpcException]):
+        """
+        Sends error to the client for displaying.
+
+        By default this fucntion does not handle LSP request errors. This is because LSP requests
+        require direct responses and so already have a mechanism for including unexpected errors
+        in the response body.
+
+        All other errors are "out of band" in the sense that the client isn't explicitly waiting
+        for them. For example diagnostics are returned as notifications, not responses to requests,
+        and so can seemingly be sent at random. Also for example consider JSON RPC serialization
+        and deserialization, if a payload cannot be parsed then the whole request/response cycle
+        cannot be completed and so one of these "out of band" error messages is sent.
+
+        These "out of band" error messages are not a requirement of the LSP spec. Pygls simply
+        offers this behaviour as a recommended default. It is perfectly reasonble to override this
+        default.
+        """
+
+        if isinstance(source, FeatureRequestError):
```

Comment:
Since the error classes rather than instances are being passed to `source`, I believe this check should be `issubclass` instead